src/commands.cpp

```diff
@@ -1958,43 +1958,57 @@ CommandGetUA::CommandGetUA(MegaClient* client, const char* uid, const char* an)
 
 void CommandGetUA::procresult()
 {
+    TLVstore *tlvRecords;
+    string tlvContainer;
+
     if (client->json.isnumeric())
     {
         error e = (error)client->json.getint();
 
 #ifdef USE_SODIUM
-        if ((e == API_ENOENT) && (user->userhandle == client->me)
-                && ((priv && strncmp(attributename.c_str(), "prEd255", 7))
-                        || (!priv && strncmp(attributename.c_str(), "puEd255", 7))))
+        if ((e == API_ENOENT) && (user->userhandle == client->me))
         {
-            // We apparently don't have Ed25519 keys, yet. Let's make 'em.
-            if(!client->inited25519())
+            // If keyring is not found, create keypairs and send them to MEGA
```

Comment:
I think it might be a good idea to extract processing for certain private and public attributes that require special treatment (keyring, pub keys, signatures) for sanity checks/"healing" to an external function to not inflate and overcomplicate this function.
But that's more an issue for further refactoring, the procedure here seems good in general.